### PR TITLE
GraphQL: Made alias field lookup case insensitive

### LINKED ISSUE #6018: Cannot using Where query from GraphQL because columns case
It's happened with the default parts (Autoroute, Alias, Localization) with postgres@10 and OC dev channel.
Somewhat related to this https://github.com/OrchardCMS/OrchardCore/issues/3684
For e.g.:
```
query MyQuery {
  blogPost(where: { path: "some-path" }) {
    displayText
  }
}
```
It throws this error:

> GraphQL.ExecutionError: Error trying to resolve something.\n ---> Npgsql.PostgresException (0x80004005): 42703: column AutoroutePartIndex.path does not exist\n   at Npgsql.NpgsqlConnector.<>c__DisplayClass160_0.<<DoReadMessage>g__ReadMessageLong|0>d.MoveNext()\n--- End of stack trace from previous location where exception was thrown ---\n   at Npgsql.NpgsqlConnector.<>c__DisplayClass160_0.<<DoReadMessage>g__ReadMessageLong|0>d.MoveNext()\n--- End of stack trace from previous location where exception was thrown ---\n   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming)\n   at Npgsql.NpgsqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)\n   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)\n   at Dapper.SqlMapper.QueryAsync[T](IDbConnection cnn, Type effectiveType, CommandDefinition command) in C:\\projects\\dapper\\Dapper\\SqlMapper.Async.cs:line 419\n   at YesSql.Store.ProduceAsync[T](WorkerQueryKey key, Func`2 work, Object[] args)\n   at YesSql.Services.DefaultQuery.Query`1.ListImpl()\n   at OrchardCore.ContentManagement.GraphQL.Queries.ContentItemsFieldType.Resolve(ResolveFieldContext context) in C:\\projects\\orchardcore\\src\\OrchardCore\\OrchardCore.ContentManagement.GraphQL\\Queries\\ContentItemsFieldType.cs:line 104\n   at OrchardCore.Apis.GraphQL.Resolvers.LockedAsyncFieldResolver`1.Resolve(ResolveFieldContext context) in C:\\projects\\orchardcore\\src\\OrchardCore\\OrchardCore.Apis.GraphQL.Abstractions\\Resolvers\\LockedAsyncFieldResolver.cs:line 25\n   at GraphQL.Execution.ExecutionStrategy.ExecuteNodeAsync(ExecutionContext context, ExecutionNode node)\n  Exception data:\n    Severity: ERROR\n    SqlState: 42703\n    MessageText: column AutoroutePartIndex.path does not exist\n    Hint: Perhaps you meant to reference the column \"AutoroutePartIndex.Path\".\n    Position: 371\n    File: parse_relation.c\n    Line: 3283\n    Routine: errorMissingColumn\n   --- End of inner exception stack trace ---

```
      {
        "code": "POSTGRES",
        "data": {
          "Severity": "ERROR",
          "InvariantSeverity": "ERROR",
          "SqlState": "42703",
          "MessageText": "column AutoroutePartIndex.path does not exist",
          "Hint": "Perhaps you meant to reference the column \"AutoroutePartIndex.Path\".",
          "Position": 371,
          "File": "parse_relation.c",
          "Line": "3283",
          "Routine": "errorMissingColumn"
        }
      }
```

Should I add an alias here for the column name:
https://github.com/OrchardCMS/OrchardCore/blob/cc53e491f9747277813ca2f72f42b5396cbe6d04/src/OrchardCore.Modules/OrchardCore.Autoroute/GraphQL/AutoroutePartIndexAliasProvider.cs#L9-L17
Or look at this function for clues: https://github.com/OrchardCMS/OrchardCore/blob/cc53e491f9747277813ca2f72f42b5396cbe6d04/src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Predicates/PredicateQuery.cs#L55-L65

Thanks,

### PR DESCRIPTION
**Pull request by carlwoodhouse**
Fixes #6018 

I probably need to test this in postgres a bit more; it's a shame we have no postgres integration tests really.

But in a nutshell the querypredicate builder for where inputs no longer assumes the casing of columns for index aliases, it will always lookup the aliased index and use the explicit column casing from the index; the same way the contentitemindex lookup worked which we also used to fix a postgres issue.

Currently i've purposely made it backwards compatible so that we dont break anyones existing integrations that havent registered property providers for custom indexes (the fallback still fail in postgres)

### CODE REVIEW

File: src/OrchardCore/OrchardCore.ContentManagement.GraphQL/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -25,9 +27,17 @@ public static IServiceCollection AddContentGraphQL(this IServiceCollection servi
 
             services.AddOptions<GraphQLContentOptions>();
 
+            services.AddSingleton(typeof(IIndexPropertyProvider), typeof(IndexPropertyProvider<ContentItemIndex>));
```
Comment (Jetski5822) [inline src/OrchardCore/OrchardCore.ContentManagement.GraphQL/ServiceCollectionExtensions.cs:30]: services.AddSingleton<IIndexPropertyProvider, IndexPropertyProvider<ContentItemIndex>>()

File: src/OrchardCore/OrchardCore.ContentManagement.GraphQL/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -25,9 +27,17 @@ public static IServiceCollection AddContentGraphQL(this IServiceCollection servi
...
+
             return services;
         }
 
+        public static void AddWhereInputIndexPropertyProvider<IIndexType>(this IServiceCollection services)
+        where IIndexType : MapIndex
+        {
+            services.AddSingleton(typeof(IIndexPropertyProvider), typeof(IndexPropertyProvider<IIndexType>));
```
Comment (Jetski5822) [inline src/OrchardCore/OrchardCore.ContentManagement.GraphQL/ServiceCollectionExtensions.cs:38]: same as above

File: src/OrchardCore/OrchardCore.ContentManagement.GraphQL/ServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -25,9 +27,17 @@ public static IServiceCollection AddContentGraphQL(this IServiceCollection servi
 
             services.AddOptions<GraphQLContentOptions>();
 
+            services.AddSingleton(typeof(IIndexPropertyProvider), typeof(IndexPropertyProvider<ContentItemIndex>));
+
             return services;
         }
 
+        public static void AddWhereInputIndexPropertyProvider<IIndexType>(this IServiceCollection services)
+        where IIndexType : MapIndex
```
Comment (Jetski5822) [inline src/OrchardCore/OrchardCore.ContentManagement.GraphQL/ServiceCollectionExtensions.cs:36]: spacing

File: src/OrchardCore/OrchardCore.ContentManagement.GraphQL/ServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -40,5 +50,7 @@ public static IServiceCollection AddContentGraphQL(this IServiceCollection servi
         {
             services.AddTransient<IGraphQLFilter<TObjectTypeToFilter>, TFilterType>();
         }
+
+
```
Comment (Jetski5822) [inline src/OrchardCore/OrchardCore.ContentManagement.GraphQL/ServiceCollectionExtensions.cs:54]: delete white space

**Comment by Jetski5822** [comment]
Left a few little comments for you @carlwoodhouse ;)

File: src/OrchardCore/OrchardCore.ContentManagement.GraphQL/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (carlwoodhouse) [reply src/OrchardCore/OrchardCore.ContentManagement.GraphQL/ServiceCollectionExtensions.cs:30]: oops think that was left over in that style from when the interface was generic too

**Review by Jetski5822 — APPROVED** [review]

**Comment by Jetski5822** [comment]
Approved - Good to merge from me

**Pull request merged**